shared/constants/metametrics.ts

```diff
@@ -0,0 +1,637 @@
+  OnboardingWalletSecurityStarted = 'SRP Backup Selected',
+  OnboardingWalletSecuritySkipInitiated = 'SRP Skip Backup Selected',
+  OnboardingWalletSecuritySkipConfirmed = 'SRP Backup Skipped',
+  OnboardingWalletSecuritySkipCanceled = 'SRP Skip Backup Canceled',
+  OnboardingWalletSecurityPhraseRevealed = 'SRP Revealed',
+  OnboardingWalletSecurityPhraseWrittenDown = 'SRP Backup Confirm Display',
+  OnboardingWalletSecurityPhraseConfirmed = 'SRP Backup Confirmed',
+  OnboardingWalletCreationComplete = 'Wallet Created',
+  OnboardingWalletSetupComplete = 'Application Opened',
+  OnboardingWalletAdvancedSettings = 'Settings Updated',
+  OnboardingWalletImportAttempted = 'Wallet Import Attempted',
+  OnboardingWalletVideoPlay = 'SRP Intro Video Played',
+  OnboardingTwitterClick = 'External Link Clicked',
+}
+
+export enum MetaMetricsEventAccountType {
+  Default = 'metamask',
+  Imported = 'imported',
+  Hardware = 'hardware',
+}
+
+export enum MetaMetricsEventAccountImportType {
+  Json = 'json',
+  PrivateKey = 'private_key',
+  Srp = 'srp',
+}
+
+export enum MetaMetricsEventCategory {
+  Accounts = 'Accounts',
+  App = 'App',
+  Auth = 'Auth',
+  Background = 'Background',
+  // The TypeScript ESLint rule is incorrectly marking this line.
+  /* eslint-disable-next-line @typescript-eslint/no-shadow */
+  Error = 'Error',
+  Footer = 'Footer',
+  Home = 'Home',
+  InpageProvider = 'inpage_provider',
+  Keys = 'Keys',
+  Messages = 'Messages',
+  Navigation = 'Navigation',
+  Network = 'Network',
+  Onboarding = 'Onboarding',
+  Phishing = 'Phishing',
+  Retention = 'Retention',
+  Settings = 'Settings',
+  Snaps = 'Snaps',
+  Swaps = 'Swaps',
+  Transactions = 'Transactions',
+  Wallet = 'Wallet',
+  Desktop = 'Desktop',
+}
+
+export enum MetaMetricsEventExternalLinkType {
+  TransactionBlockExplorer = 'Transaction Block Explorer',
+  BlockExplorer = 'Block Explorer',
+  AccountTracker = 'Account Tracker',
+  TokenTracker = 'Token Tracker',
+}
+
+export enum MetaMetricsEventKeyType {
+  Pkey = 'private_key',
+  Srp = 'srp',
+}
+
+// NOTE: This doesn't seem to be used at all
+export enum MetaMetricsEventOnrampProviderType {
+  Coinbase = 'coinbase',
+  Moonpay = 'moonpay',
+  Wyre = 'wyre',
+  Transak = 'transak',
+  SelfDeposit = 'direct_deposit',
+}
+
+export enum MetaMetricsEventNetworkSource {
+  CustomNetworkForm = 'custom_network_form',
+  PopularNetworkList = 'popular_network_list',
+  Dapp = 'dapp',
+}
+
+export enum MetaMetricsEventSwapsSource {
+  MainView = 'Main View',
+  TokenView = 'Token View',
+}
+
+export enum MetaMetricsEventTokenSource {
+  Custom = 'custom',
+  Dapp = 'dapp',
+  Detected = 'detected',
+  List = 'list',
+}
+
+export enum MetaMetricsEventTransactionSource {
+  Dapp = 'dapp',
+  User = 'user',
+}
+
+export enum MetaMetricsEventLocation {
+  TokenDetails = 'token_details',
+  TokenDetection = 'token_detection',
+  TokenMenu = 'token_menu',
+}
+
+export enum MetaMetricsEventUiCustomizationPropertyValue {
```

Comment:
Previously this enum was nested in another object (`METAMETRIC_KEY_OPTIONS`). Considering that this object was being used for only one thing, I didn't think it was worth keeping that structure. Also, as above, enum values must be simple so this enum follows the same pattern as the enums above.